### PR TITLE
YJIT: limit size of call count stats dict

### PR DESCRIPTION
**Pull request by maximecb**
Someone reported two weeks ago that logs were getting bloated because the ISEQ and C call count dicts were huge, since they include all of the call sites. I wrote code on the Rust size to limit the size of the dict to avoid this problem. The size limit is hardcoded at 20, but I figure this is probably fine?

### CODE REVIEW

File: yjit/src/stats.rs [hunk 1]
```diff
@@ -802,16 +802,31 @@ fn rb_yjit_gen_stats_dict(context: bool) -> VALUE {
...
                     let count = call_counts[*idx];
+                    pairs.push((name, count));
+                }
+
+                // Sort the vectors by decreasing call counts
+                pairs.sort_by_key(|e| -(e.1 as i64));
+
+                // Cap the number of counts reported to avoid
+                // bloating log files, etc.
+                pairs = pairs[0..20].to_vec();
```
Comment (k0kubun) [inline yjit/src/stats.rs:824]: This seems to crash when it has less than 20 elements. I guess you can use `pairs.iter().take(20)`?

**Comment by k0kubun** [comment]
This would break the calculation of `top_n_pct` https://github.com/ruby/ruby/blob/1eff5a98f1c83f37ca2daac1c52f3eb84f395acd/yjit.rb#L420-L421. Is that intentional?

**Comment by maximecb** [comment]
> This would break the calculation of `top_n_pct`.
> 
> https://github.com/ruby/ruby/blob/1eff5a98f1c83f37ca2daac1c52f3eb84f395acd/yjit.rb#L420-L421
> 
> Is that intentional?

AFAIK it doesn't break `top_n_pct`, because `num_calls` is not calculated based on the dict, we're gathering that count separately?

**Comment by k0kubun** [comment]
🤦 You're right, I was reading it wrong. I'm sorry.

**Pull request merged**